src/codegen/codeparser.h

```diff
@@ -58,98 +62,132 @@ class Function {
+
+	wxString m_userIncludeFilePreProc;
+	wxString m_userSourceFilePreProc;
+	wxString m_className;
+	wxString m_userHeaderFileMembers;
+	wxString m_userSourceFileTrailing;
+
+
+	ArrayCppFunction m_functionList;
+	int lastRetFunc = 0;
+
 public:
 	/** constructor */
-	CodeParser() {
+	CppCodeParser() {
 	}
 
-	~CodeParser() {
+	CppCodeParser(wxString &headerFileName, wxString &sourceFileName) {
+		m_headerFileName = headerFileName;
+		m_sourceFileName = sourceFileName;
 	}
 
-	///////////////////////////////////////////////////////////////////
-
-	/** returns all user header include code before the class declaration */
-	wxString GetUserIncludes() {
-		return m_userInclude;
+	~CppCodeParser() {
+		m_functionList.Clear();
 	}
 
-	/** returns user class members */
-	wxString GetUserMembers() {
-		return m_userMemebers;
-	}
+	/** c++ Parser */
 
-	/** returns the Documentation of a function by name */
-	wxString GetFunctionDocumentation(wxString function);
+	/** opens the header and source files,  'className' is the Inherited class */
+	void ParseFiles(wxString &className);
 
-	/** returns the contents of a function by name and then removes it from the list of remaining
-	 * functions */
-	wxString GetFunctionContents(wxString function);
+	/** extracts the contents of the files.  take the the entire contents of both files in string
+	 * form */
+	void ParseCode(wxString &header, wxString &source);
 
-	/** returns all ramaining functions including documentation as one string.
-	 this may rearange functions, but should keep them intact */
-	wxString GetRemainingFunctions();
+	/** extracts all user preprocessor definitions before the class declaration in include file*/
+	void ParseHeaderFilePreProc(wxString &code);
 
-	wxString GetTrailingCode() {
-		return m_trailingCode;
-	}
+	/** extracts the contents of the inherited class declaration */
+	void ParseHeaderFileClass(wxString &code);
 
-protected:
-	wxString m_userInclude;
-	wxString m_className;
-	wxString m_userMemebers;
-	wxString m_trailingCode;
+	/** extracts user created class members **/
+	void ParseHeaderFileUserMembers(wxString &code);
 
-	FunctionMap m_functions;
-	funcIterator m_functionIter;
-};
 
-/** parses the source and header files for all code added to the generated */
-class CCodeParser : public CodeParser {
-private:
-	wxString m_hFile;
-	wxString m_cFile;
 
-public:
-	/** constructor */
-	CCodeParser() {
-	}
+	/** extracts all user preprocessor definitions before the class declaration in source file*/
+	void ParseSourceFilePreProc(wxString &code);
 
-	CCodeParser(wxString headerFileName, wxString sourceFileName) {
-		m_hFile = headerFileName;
-		m_cFile = sourceFileName;
-	}
+	/** extracts each function in source File**/
+	void ParseSourceFileFunctions(wxString &code);
+
+	wxString ParseBrackets(wxString &code, int &functionStart);
+
+	// Accessor Functions
+	////////////////////////
+	wxString GetUserHeaderFilePreProc() {return m_userIncludeFilePreProc;}
```

Comment:
For simple accessors one-liners are ok but please add whitepace inside the braces